### ISSUE TITLE
Support for updating axis label positions asynchronously

### ISSUE THREAD

**Issue by benelliott** [highcharts/highcharts#18175]
# Description of the feature

Hi, I am trying to figure out how to update axis label content asynchronously (i.e., outside of the body of the `formatter` function) but can't figure out how to then get Highcharts to reflow the label content to fit the new values.

The approach I am trying to take is similar to the one suggested in https://github.com/highcharts/highcharts/issues/6824,
i.e. get a handle to the `label` SVG element and then use `.attr({text: '... new value...'})`.

I have got this working as can be seen in this Fiddle: https://jsfiddle.net/pkwx5a4d/

However, I have been unable to find a way to force Highcharts to recompute the axis label sizes and shift things around accordingly so they fit without overlapping.  I've tried various combinations of `reflow()`, `redraw()`, `getBbox(true)` with no luck.

Is there any way to achieve this?

Thanks!

# Library related to the feature

# Proof of Concept/Live example for the feature

----

**You can vote for this feature by adding a thumbs-up reaction to this post.**

**Comment by karolkolodziej** [comment]
Hi @benelliott!

I managed to work around this by returning a placeholder in the label formatter. 
Then you can change that text as  I did in the [demo](https://jsfiddle.net/BlackLabel/2mq3o0x1/).
To solve the issue with proper spacing you have to slightly change the container size in order to properly reflow the chart.

**Comment by benelliott** [comment]
Hi @karolkolodziej, thanks for the reply. With your approach, I can't seem to revert the container back to its old size without the label being cropped again?

**Comment by karolkolodziej** [comment]
That is correct, with this approach I was not able to restore the initial width.

**Comment by benelliott** [comment]
Thanks, in that case this is not a viable workaround for me unfortunately (I will be updating the axis labels repeatedly).

**Comment by benelliott** [comment]
Bump
